k-distribution/src/main/scripts/lib/kit/utils.py

```diff
@@ -0,0 +1,68 @@
+from collections import Counter
+from typing import Callable, Type, TypeVar
+
+from pyk.kast import KApply, KInner, KRewrite, KSequence, KVariable
+from pyk.kastManip import (
+    countVarOccurances,
+    rewriteAnywhereWith,
+    traverseTopDown,
+)
+
+KI = TypeVar('KI', bound=KInner)
+
+
+def eliminate_generated_top(term: KInner) -> KInner:
+    rule = KApply('<generatedTop>', [KVariable('CONFIG'), KVariable('_')]), KVariable('CONFIG')
+    return rewriteAnywhereWith(rule, term)
```

Comment:
What do you think of being able to write instead:

```suggestion
def eliminate_generated_top(term: KInner) -> KInner:
    rule = KRewrite(KApply('<generatedTop>', [KVariable('CONFIG'), KVariable('_')]), KVariable('CONFIG'))
    return rewriteAnywhereWith(rule, term)
```

Not a huge difference, basically have the various `rewrite*` functions take a `KRewrite` as their first argument, instead of just a tuple of LHS/RHS. It seems appropriate, and more structured.

Reply:
Yes, I think this is a good idea. Such rewrite / replace functions can even be methods of the `KRewrite` class.